### DISCUSSION TITLE
Reverse sync - deployment aid

### DISCUSSION THREAD

**Discussion by robbykrlos** [worksome/envy#13] [Q&A] [answered]
Hi,

I'm facing the following challenge, and was wondering if using this package will help.

Scenario: 
Application aged well, stable, deployed on 5 servers.
I have a big feature release that I need to deploy. Since we do not use cloud solution, and have intranet servers, we have some deployment scripts that pulls the latest code, and runs some commands like, migrate, optimize, etc.

Problem is that if my new features contains also some new `.env` configuration items needed for deployment, I need to add them manually on each server in the .env.

I was looking for a way, similar to php artisan migrate, where I can mention something like:
```php
$this->env->createOrUpdate('APP_VERSION', '4.2.0')
```

So that in the end, whenever I run my deployment scripts I can maybe call `php artisan env:update` and get all my servers updated with correct configuration.

I was thinking this can be achieved using envy by reverse syncing .env with .env.example, and whatever is new in the .env.example (which is usually not ignored by git) will be transferred into the new environment's .env.

What do you think about this?

Thanks!

**Comment by olivernybroe** [comment, marked as answer]
Hi, thanks for suggesting a new feature. 

So the issue I see with a migration for env variables is that the values often are private and should not be in VCS.

I got two other solutions which I think you could consider doing.
1. In your config files where you use the env variables, set a default value to the value you want on all the servers.
2. In your servers rename your `.env` file to `.env.production` and on each deploy rename `.env.example` to `.env`. That way both of them are loaded, but `.env.production` values will always have precedence. 

Would any of these solutions work for you instead of having a new feature in this package? :)

**Reply by robbykrlos** [reply]
@olivernybroe I think point 2. is the closest - I did not know that they have precedence and can be loaded partially and some configs will override the others. Clear now. No need for a feature.

**Reply by olivernybroe** [reply]
Awesome, glad I could help!
